### PR TITLE
Bugfix trello 2340 checks stuck in running state

### PR DESCRIPTION
**Pull request by ikseek**

### CODE REVIEW

File: eyes_selenium/applitools/selenium/visual_grid/dom_snapshot_script.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
 MAX_CHUNK_BYTES_GENERIC = 50 * 1024 * 1024
 SCRIPT_POLL_INTERVAL_MS = 1000
 
+CORS_IFRAME_PROCESSING_ENABLED = True
```
Comment (skhalymon) [inline eyes_selenium/applitools/selenium/visual_grid/dom_snapshot_script.py:28]: Probably need clarification about this API. Should it be in Configuration? @danielputerman

File: eyes_selenium/applitools/selenium/visual_grid/resource_cache.py [hunk 2]
```diff
@@ -76,3 +82,58 @@ def fetch_and_store(self, url, fetch_function, force=False):
     def process_all(self):
         with self.lock:
             return [self[r_url] for r_url in self]
+
+
+class PutCache(object):
+    def __init__(self):
+        if sys.version_info[:2] >= (3, 6):
```
Comment (skhalymon) [inline eyes_selenium/applitools/selenium/visual_grid/resource_cache.py:89]: Would be cleaner to omit else block?

File: eyes_selenium/applitools/selenium/visual_grid/dom_snapshot_script.py (hunk 1 above)
Comment (ikseek) [reply eyes_selenium/applitools/selenium/visual_grid/dom_snapshot_script.py:28]: As far as I know, API is being discussed. I left it here temporarily to aid in stabilisation of SDC suite. It should be replaced with API for sure when it's ready.

File: eyes_selenium/applitools/selenium/visual_grid/resource_cache.py [hunk 3]
```diff
@@ -108,25 +109,22 @@ def put(
...
+            all_resources = [full_request_resources.get(url) for url in urls]
+            put_resources = [
+                r for r in all_resources if force or r.hash not in self._sent_hashes
+            ]
+            self._sent_hashes.update(r.hash for r in put_resources)
+            futures = self._executor.map(
+                partial(eyes_connector.render_put_resource, render_id),
+                put_resources,
+            )
+            self._currently_uploading.append(futures)
```
Comment (skhalymon) [inline eyes_selenium/applitools/selenium/visual_grid/resource_cache.py:120]: Should be `self._currently_uploading.append` replaced with `extend()`? In the result, we could use `as_completed(futures)`

**Pull request merged**